### PR TITLE
[Relay] Fix an adaptive_max_pool1d operator conversion bug

### PR DESCRIPTION
**Pull request by haoyang9804**
This is a code patch for fixing this issue: https://github.com/apache/tvm/issues/15004
Please refer to my comments in this link for the root cause of this issue.

### CODE REVIEW

**Comment by haoyang9804** [comment]
cc @jikechao

**Review by jikechao — CHANGES_REQUESTED** [review]
Thanks for your fixing, LGTM. Please add a test!

**Review by jikechao — APPROVED** [review]

**Comment by jikechao** [comment]
@echuraev @Hzfengsy @vvchernov 
Please recheck it.

**Comment by Hzfengsy** [comment]
Please add a regression test for it

**Comment by haoyang9804** [comment]
> From one hand LGTM, from another one it confuses me due to fix for specific op was done on very high level (and CI test was constructed for specific op not for all cases). It looks like it has been assumed that outputs exist at any `ret_name in ret_names`. I need time to analyze it by my-self or explanation what is wrong with the initial assumption.

I agree that "fix for specific op was done on very high level". So I make the fix only specific to the buggy pytorch op. This is only a workaround currently. I believe there are some deeper issues make this bug happen.

**Comment by jikechao** [comment]
cc @vvchernov

File: python/tvm/relay/frontend/pytorch.py [hunk 1]
```diff
@@ -4291,7 +4291,11 @@ def _handel_nested_input(inputs):
 
             self.current_op.pop()
 
-        return [_wrap_const(outputs[ret_name]) for ret_name in ret_names]
+        return [
+            _wrap_const(outputs[ret_name])
+            for ret_name in ret_names
+            if ret_name != "aten::adaptive_max_pool1d_0_1"
```
Comment (vvchernov) [inline python/tvm/relay/frontend/pytorch.py:4297]: LGTM, but please set TODO somewhere near. Due to it is still workaround for specific op on high-level API and we do not miss it for more correct fix in the future.

**Review by vvchernov — CHANGES_REQUESTED** [review]

File: python/tvm/relay/frontend/pytorch.py (hunk 1 above)
Comment (haoyang9804) [reply python/tvm/relay/frontend/pytorch.py:4297]: I found that all TODO in python requires @someone. In this issue, who should I @?

Comment (vvchernov) [reply python/tvm/relay/frontend/pytorch.py:4297]: Of course your-self) In general It does not matter due to it does not mean that the man who wrote TODO should fix it, it is a tag for anybody to understand that there is some issue

**Review by vvchernov — APPROVED** [review]

**Review by echuraev — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**